src/main/java/com/redislabs/ReadSource.java

```diff
@@ -82,9 +82,11 @@ public void foreach(KeysReaderRecord r) throws Exception {
           GearsBuilder.overrideReply(error);
           throw new Exception(error);
         }
-        command.add(propKey);
-        Object value = e.getValue();
-        command.add(pd.convertToStr(value));
+        if (value != null) {
+          command.add(propKey);
+          Object value = e.getValue();
```

Comment:
Get this line out of the if statement?

Reply:
Also maybe we do want to add an empty value on this field?

Reply:
Don't think so, there is a different between null and empty string

Reply:
Look at the comment bellow then, we should not return this field at all then ...

Reply:
right this is why I added `command.add(propKey);` to the if part